### PR TITLE
evalengine: Implement BIN, OCT & CHAR functions

### PR DESCRIPTION
**Pull request by dbussink**
This implements `BIN(<val>)` as `CONV(<val>, 10, 2)`. It also fixes an edge case in uint64 parsing to match MySQL behavior that was exposed with new tests added for `BIN`. 

Also adds `OCT(<val>)` as `CONV(<val>, 10, 8)` and adds `CHAR` too. 

## Related Issue(s)

Part of #9647 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/mysql/fastparse/fastparse.go [hunk 1]
```diff
@@ -45,6 +45,22 @@ func ParseUint64(s string, base int) (uint64, error) {
...
+	minus := s[i] == '-'
+	if minus {
+		if !allowNeg {
+			return 0, fmt.Errorf("cannot parse uint64 from %q", s)
+		}
+		i++
+		if i >= uint(len(s)) {
+			return 0, fmt.Errorf("cannot parse uint64 from %q", s)
+		}
+	}
```
Comment (vmg) [inline go/mysql/fastparse/fastparse.go:62]: Very nice catch. Also:

image

Also: I think the `bool` flag to the API makes it a bit sharp/footgun, and it makes the body confusing. Could you please split this this in two functions? I think there's a pretty clean way to do this where all the parsing code is extracted into its own internal function, and then we have two entrypoints, and the `uValue` complement is just performed on the return value for the function that handles negatives. Does that make sense? LMK and I can try the split myself.

**Review by vmg — APPROVED** [review]

**Review by GuptaManan100 — APPROVED** [review]
LGTM
